azure/services/scalesets/scalesets.go

```diff
@@ -85,86 +79,57 @@ func (s *Service) Reconcile(ctx context.Context) (retErr error) {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "scalesets.Service.Reconcile")
 	defer done()
 
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
 	if err := s.validateSpec(ctx); err != nil {
```

Comment:
would it make sense to move this into `Parameters` in spec.go? that's what we do for VMs already, and that way we wouldn't have to call `s.Scope.ScaleSetSpec(ctx)` twice

Reply:
I think the intention here was to do the validation before we call `client.Get()` and `client.ListInstances()`. The main thing for me is that `Parameters()` is already gigantic and pretty difficult to maintain/test, especially the different control flow possibilities. WDYT?